### PR TITLE
feat: Add manila csi

### PR DESCRIPTION
**Pull request by okozachenko1203**
- This PR includes manila-csi-plugin deployment. This PR extends `magnum.conf` by adding `[manila_client]` section for keystone authtoken. So magnum driver can configure manila_client and use this to interact with sharev2 api.
- `magnum` doesn't support `manila` intergration so I override `magnum.common.clients.OpenStackClients` to include `manila` method which returns the manila client.
- This PR also makes cinder-csi-plugin and manila-csi-plugin as conditional.  i.e. even if labels (`cinder_csi_enabled`, `manila_csi_enabled`) are set as `true`, if the underlying openstack has no enabled endpoints for cinder and manila, it will not install those plugins.

fix: https://github.com/vexxhost/magnum-cluster-api/issues/79

### CODE REVIEW

File: magnum_cluster_api/utils.py [hunk 1]
```diff
@@ -233,3 +236,56 @@ def delete_loadbalancers(ctx, cluster):
...
+        manila_svc = keystone.client.services.list(type="sharev2")
+    except Exception:
+        LOG.exception("Failed to list services")
+        raise exception.ServicesListFailed()
+
+    if manila_svc and manila_svc[0].enabled:
+        return True
+
+    LOG.info("There is no sharev2 service enabled in the cloud.")
+    return False
```
Comment (okozachenko1203) [inline magnum_cluster_api/utils.py:277]: to detect services in the cloud

**Review by okozachenko1203 — COMMENTED** [review]
@mnaser i added comments where requires more attention.

**Comment by mnaser** [comment]
@okozachenko1203 I previously built a tool here which will 'sync' all the manifests and apply all the changes that we use -- https://github.com/vexxhost/magnum-cluster-api/blob/main/tools/sync-csi-manifests

There are other sync jobs too.  I think what I would like is that maybe we extend that script, and we also create a job that runs this script in CI and uses git to make sure that no changes happen, which means that no one made changes that were not implmented in this script and we can 'reliabely' get the manifests by running that script.

The potential second step to this is adding this script into the build script so that the manifestis not bundled in the source code but added in build time like we do for charts.

**Comment by okozachenko1203** [comment]
> @okozachenko1203 I previously built a tool here which will 'sync' all the manifests and apply all the changes that we use -- https://github.com/vexxhost/magnum-cluster-api/blob/main/tools/sync-csi-manifests
> 
> There are other sync jobs too. I think what I would like is that maybe we extend that script, and we also create a job that runs this script in CI and uses git to make sure that no changes happen, which means that no one made changes that were not implmented in this script and we can 'reliabely' get the manifests by running that script.
> 
> The potential second step to this is adding this script into the build script so that the manifestis not bundled in the source code but added in build time like we do for charts.

I think we can consider keeping diff/patch files instead of writing py script.

**Pull request merged**